src/agent0/chainsync/db/hyperdrive/chain_to_db.py

```diff
@@ -100,3 +113,325 @@ def data_chain_to_db(interface: HyperdriveReadInterface, block: BlockData, sessi
+    # TODO we have a test for initializer of the pool, but we need to implement
+    # wallet to wallet transfers of tokens in the interactive interface for a full test
+    transfer_events_trx_hash = unique_events_per_transaction[
+        unique_events_per_transaction["nunique"] < 2
+    ].reset_index()["transactionHash"]
+    transfer_events_df = events_df[events_df["transactionHash"].isin(transfer_events_trx_hash)].copy()
+    if len(transfer_events_df) > 0:
+        # Expand the args dict without losing the args dict field
+        # json_normalize works on series, but typing doesn't support it.
+        args_columns = pd.json_normalize(transfer_events_df["args"])  # type: ignore
+        transfer_events_df = pd.concat([transfer_events_df, args_columns], axis=1)
+        # We apply the decode function to each element, then expand the resulting
+        # tuple to multiple columns
+        transfer_events_df["token_type"], transfer_events_df["maturityTime"] = zip(
+            *transfer_events_df["id"].astype(int).apply(decode_asset_id)
+        )
+        # Convert token_type enum to name
+        transfer_events_df["token_type"] = transfer_events_df["token_type"].apply(lambda x: AssetIdPrefix(x).name)
+        # Convert maturity times of 0 to nan to match other events
+        transfer_events_df.loc[transfer_events_df["maturityTime"] == 0, "maturityTime"] = np.nan
+        # Set token id, default is to set it to the token type
+        transfer_events_df["token_id"] = transfer_events_df["token_type"]
+        # Append the maturity time for longs and shorts
+        long_or_short_idx = transfer_events_df["token_type"].isin(["LONG", "SHORT"])
+        transfer_events_df.loc[long_or_short_idx, "token_id"] = (
+            transfer_events_df.loc[long_or_short_idx, "token_type"]
+            + "-"
+            + transfer_events_df.loc[long_or_short_idx, "maturityTime"].astype(str)
+        )
+
+        # Set the trader of this transfer
+        transfer_events_df["trader"] = wallet_addr
+
+        # See if it's a receive or send of tokens
+        send_idx = transfer_events_df["from"] == wallet_addr
+        receive_idx = transfer_events_df["to"] == wallet_addr
+        # Set the token delta based on send or receive
+        transfer_events_df.loc[send_idx, "token_delta"] = -transfer_events_df.loc[send_idx, "value"].apply(
+            lambda x: Decimal(x) / Decimal(1e18)  # type: ignore
+        )
+        transfer_events_df.loc[receive_idx, "token_delta"] = transfer_events_df.loc[receive_idx, "value"].apply(
+            lambda x: Decimal(x) / Decimal(1e18)  # type: ignore
+        )
+        # Base delta is always 0
+        transfer_events_df["base_delta"] = Decimal(0)
+
+    # Drop all transfer single events
+    events_df = events_df[events_df["event"] != "TransferSingle"].reset_index(drop=True)
+
+    # Sanity check, one hyperdrive event per transaction hash
+    assert events_df.groupby("transactionHash")["event"].nunique().all() == 1
```

Comment:
better to put this in a conditional and then raise an error

- works well with down-stream error handling
- running the program in optimized mode (by setting the `PYTHONOPTIMIZE` command line flag, `-O`) silently disables assert statements.